numpy/lib/function_base.py

```diff
@@ -3870,7 +3870,7 @@ def _quantile_ureduce_func(a, q, axis=None, out=None, overwrite_input=False,
             "'midpoint', or 'nearest'")
 
     n = np.array(False, dtype=bool)  # check for nan's flag
-    if indices.dtype == intp:  # take the points along axis
+    if np.issubdtype(indices.dtype, np.integer):  # take the points along axis
```

Comment:
Not quite sure this is right, maybe we should consider `interpolation not in ("midpoint", "linear"):` or so.

Reply:
Can you elaborate on that a little? I'm not quite sure what you mean.

Reply:
Sorry, that was silly. It was the alternative where integer inputs are not special, but than I decided that Eric was likely right and it is OK if integer `q` has somewhat special behaviour.

Reply:
OK so for this piece is there any action on my part you think I should address?

Reply:
No, I forgot just forgot to delete it. It would be the other way to make it consistent (but with different behaviour choice).